### PR TITLE
fix(components/data-manager): getDataStateUpdates no longer drops changes when peers mutate shared state

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#4099031](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/4099031)

`distinctUntilChanged` retained a live reference to the previously emitted
`SkyDataManagerState`. Several participants (the filter controller directive
and four toolbar actions) mutated that shared instance in place before
republishing, corrupting the retained "previous" value and silently
dropping real changes for properties/comparator subscribers. Also guard
against a throwing comparator or unserializable state, which previously
killed the subscription permanently and silently.

Co-Authored-By: Claude Sonnet 5 <noreply@anthropic.com>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved data grid filtering, searching, sorting, and selection updates so subscribers consistently receive the latest state.
  * Prevented previously emitted data states from being modified unexpectedly.
  * Improved resilience when state comparisons encounter invalid or unsupported data.

* **Tests**
  * Added regression coverage for consecutive filter changes, state immutability, subscriber updates, and comparison edge cases.

* **Chores**
  * Increased the production style-size budget to support larger component styles.

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/4700/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/4700/storybooks/ag-grid-storybook/)
- [data-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/4700/storybooks/data-grid-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/4700/storybooks/data-manager-storybook/)

Apps:

- [code-examples-playground](https://blackbaud.github.io/skyux-pr-preview/4700/code-examples-playground/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/4700/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/4700/playground/)

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-filters/data-manager-filter-controller.directive.spec.ts [hunk 1]
```diff
@@ -94,4 +94,61 @@ describe('SkyDataManagerFilterControllerDirective', () => {
 
     expect(spy).toHaveBeenCalled();
   });
+
+  it('should notify property-filtered subscribers of each consecutive filter change', () => {
+    // Regression test for AB#4099031: subscribers using `properties`/`comparator` filtering
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/data-manager/src/lib/modules/data-manager/data-manager-filters/data-manager-filter-controller.directive.spec.ts]: can we remove this comment? I'd rather not reference ADO tickets in our code

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts [hunk 2]
```diff
@@ -521,6 +521,10 @@ describe('SkyDataManagerToolbarComponent', () => {
     tick();
     dataManagerToolbarFixture.detectChanges();
 
+    // Re-read the current data state rather than a reference captured before the action: the
+    // toolbar now builds a new state instance for each update rather than mutating one in place.
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts]: in general I'm very wary of all the comments claude adds.  "rather than reference...toolbar now...rather than mutating" all points to implementation details of "it used to do X, but rather than X, now it does Y". I don't think those types of comments really belong in the code, personally? Comments should be limited to explaining technical decisions in the current implementation if totally necessary. If we think this comment is necessary, I think it could be rephrased in a way that doesn't sound so specific to this bug.

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts [hunk 3]
```diff
@@ -684,9 +690,11 @@ describe('SkyDataManagerToolbarComponent', () => {
       });
     }
 
-    expect(dataManagerToolbarComponent.dataState?.filterData).toEqual(
-      filterData,
-    );
+    // Re-read the current data state rather than a reference captured before the action: the
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts]: also, the comments are repetitive

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.ts [hunk 4]
```diff
@@ -217,8 +223,10 @@ export class SkyDataManagerToolbarComponent implements OnDestroy, OnInit {
 
   public searchApplied(text: string): void {
     if (this.dataState) {
-      this.dataState.searchText = text;
-      this.#dataManagerService.updateDataState(this.dataState, this.#_source);
+      // See the comment in `sortSelected` on why a new state is built rather than mutated.
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.ts]: why 😭 edit: why x4, this is in 4 places

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts (hunk 2 above)
Comment (johnhwhite) [reply libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts]: Fair. Dropped the comment.

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts (hunk 3 above)
Comment (johnhwhite) [reply libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts]: Removed.

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.ts (hunk 4 above)
Comment (johnhwhite) [reply libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.ts]: Cleared these out. 🫥

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-filters/data-manager-filter-controller.directive.spec.ts (hunk 1 above)
Comment (johnhwhite) [reply libs/components/data-manager/src/lib/modules/data-manager/data-manager-filters/data-manager-filter-controller.directive.spec.ts]: Fair. Dropped the comment.

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/4732
